table/table_iterator.h

```diff
@@ -48,22 +49,65 @@ class TableIterator : public Iterator {
   bool Valid() const override { return iter_->Valid(); }
   void SeekToFirst() override { return iter_->SeekToFirst(); }
   void SeekToLast() override { return iter_->SeekToLast(); }
-  void Seek(const Slice& target) override { return iter_->Seek(target); }
+  void Seek(const Slice& target) override {
```

Comment:
The type of key taken as input for `Seek` and `SeekForPrev` should be consistent with the type of key produced by `Iterator::key`. Can you add jni APIs for the util methods `GetInternalKeyForSeek` etc instead?

Reply:
Can't we change the definition of the TableIterator to seek and get only on the user_key instead? Isn't this a better way to define this class?

Reply:
The name `TableIterator` indicates it's iterating the raw table, and that makes internal key a better format of key for these APIs.